morpheus/app/render/main.py

```diff
@@ -78,13 +78,16 @@ def looks_like_link(s):
 
 def apply_short_links(context, click_url, click_random=30):
     shortened_link = []
+    long_links = {}
     for k, v in context.items():
         # TODO deal with unsubscribe links properly
         if k != 'unsubscribe_link' and looks_like_link(v):
             r = secrets.token_urlsafe(click_random)[:click_random]
             context[k] = click_url + r
+            long_links[f'{k}_long_link'] = v
             shortened_link.append((v, r))
-    return shortened_link
+    context.update(long_links)
+    return context, shortened_link
```

Comment:
you don't need to return `context`, it's mutable and will be modified by the function